mixins/loading-complete/loading-complete-mixin.js

```diff
@@ -1,21 +1,36 @@
-export const LoadingCompleteMixin = (superclass) => class extends superclass {
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
 
-	constructor() {
-		super();
-		this._loadingCompletePromise = new Promise(resolve => {
-			this._loadingCompleteResolve = resolve;
-		});
+export const LoadingCompleteMixin = dedupeMixin((superclass) => class extends superclass {
+
+	#loadingCompleteResolve;
+
+	// eslint-disable-next-line sort-class-members/sort-class-members
+	#loadingCompletePromise = this.getLoadingComplete === this.#getLoadingComplete
+		? new Promise(resolve => this.#loadingCompleteResolve = resolve)
+		: Promise.resolve();
```

Comment:
If there is already a `getLoadingComplete` method, don't wait for `resolveLoadingComplete` to be called.

Reply:
Quick double check around order of execution here, this is essentially sugar for constructor-level definitions right? i.e. do we expect this to evaluate to `Promise.resolve` in cases where the consumer component overrides `getLoadingComplete`, and not just in cases where mixins override it?

Reply:
Yes, it's the same thing as a constructor assignment, but I'll have to double-check the order and if this actually works with mixed use between component and mixin. I think it could fail to resolve the default promise as-is.

Reply:
Ok, that is exactly what was happening if I swapped the mixin chain around. Fixed by using `hasOwn` instead of comparing the methods.